index.js

```diff
@@ -3,53 +3,227 @@ const state = {
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "006-bananas",
       name: "bananas",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "007-bell-pepper",
       name: "bell pepper",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "008-berry",
       name: "berry",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "009-blueberry",
       name: "blueberry",
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "010-eggplant",
       name: "eggplant",
-      price: 0.35
-    }
+      price: 0.35,
+    },
   ],
-  cart: []
-};
\ No newline at end of file
+  cart: [],
+};
+
+// storeItems, cartItems and total selectors
+const storeItemsUl = document.querySelector(".store--item-list");
+const cartItemsUl = document.querySelector(".cart--item-list");
+
+// clear the stroreItems and cartItems
+function clearingStoreItems() {
+  storeItemsUl.innerText = "";
+}
+
+function clearingCart() {
+  cartItemsUl.innerText = "";
+}
+
+function renderingItemsToStore() {
+  clearingStoreItems();
+  // Create the for loop for the items in the store
+  for (let i = 0; i < state.items.length; i++) {
+    const specificFruit = state.items[i];
+
+    // section 1: create the list item like template
+    // create the list items by createelement
+    // than append li to ul
+    const storeItemsList = document.createElement("li");
+    storeItemsUl.append(storeItemsList);
+
+    // create the div
+    // set an attribute class to the div
+    const itemsDiv = document.createElement("div");
+    itemsDiv.setAttribute("class", "store--item-icon");
+
+    // create the image elements and assign the path of the image source
+    const fruitImage = document.createElement("img");
+    fruitImage.setAttribute("src", `assets/icons/${specificFruit.id}.svg`);
+    fruitImage.setAttribute("alt", "specificFruit.name");
+
+    // also create  the button of "add to cart"
+    const addToCartButton = document.createElement("button");
+    addToCartButton.innerText = "Add to cart";
+
+    // append the div to the list
+    storeItemsList.append(itemsDiv);
+    itemsDiv.append(fruitImage);
+    storeItemsList.append(addToCartButton);
+
+    // add Event listener to the button and on click event it should add the item to the cart
+    addToCartButton.addEventListener("click", () => {
+      addingFruitToCart(specificFruit);
+      renderItemsToCart();
+      // add the sum of the items
+      totalOfCart();
+    });
+  }
+}
+// renderingItemsToStore()
+
+// checking the cart if there are items or not
+let itemsAlreadyInTheCart = false;
+function checkIfItemIsInCart(item) {
+  for (i = 0; i < state.cart.length; i++) {
+    if (state.cart[i].name === item.name) {
+      itemsAlreadyInTheCart = true;
+      state.cart[i].quantity += 1;
+      break;
+    } else {
+      itemsAlreadyInTheCart = false;
+    }
+  }
+}
+
+// create the new object of items to the cart array
+function addingFruitToCart(fruit) {
```

Comment:
running the check for if the item is already in the cart before creating the new obj would be better but is only a very minor thing.